packages/ui/src/components/NumberInputV2/index.tsx

```diff
@@ -0,0 +1,373 @@
+import styled from '@emotion/styled'
+import { Icon } from '@ultraviolet/icons'
+import type { ForwardedRef, InputHTMLAttributes, ReactNode } from 'react'
+import {
+  createRef,
+  forwardRef,
+  useCallback,
+  useId,
+  useImperativeHandle,
+  useMemo,
+} from 'react'
+import { Button } from '../Button'
+import { Stack } from '../Stack'
+import { Text } from '../Text'
+import { Tooltip } from '../Tooltip'
+
+const SIZES = {
+  small: '34px',
+  medium: '40px',
+  large: '48px',
+}
+
+type Sizes = keyof typeof SIZES
+
+const SideContainer = styled('div', {
+  shouldForwardProp: prop => !['size'].includes(prop),
+})<{ size: Sizes }>`
+  &[data-position='left'] {
+    border-right: 1px solid ${({ theme }) => theme.colors.neutral.border};
```

Comment:
May be you could do, this way you are sure SideContainer follow parent border-color including hover/focus cases
```css
   border-right: 1px solid;
   border-color: inherit;
```

PS : I didn't find a way to one line this `border-right: 1px solid "inherit";`

Reply:
@fabienhebert good idea im gonna do this